### DISCUSSION TITLE
PNG output works, PDF produces blank page

### DISCUSSION THREAD

**Discussion by madduck** [ImageMagick/ImageMagick#5039] [Help] [answered]
Hey, I am trying to auto-generate PDF files I can use to watermark with `pdftk input.pdf background watermark.pdf output.pdf`. Thanks to [help received by some people here](https://github.com/ImageMagick/ImageMagick/discussions/5036), I came up with the following command to generate the `watermark.pdf` file:

```
convert -size 4330x -density 300x300 -background transparent -gravity center -fill grey label:"This is a long text" -trim -rotate -54 +repage -resize $((2480*.8))x$((3508*.8)) -extent 2480x3508 /tmp/watermark.png

convert /tmp/watermark.png /tmp/watermark.pdf
```

The problem is that this only works if I write the PNG file first, and convert it afterwards. If I try to use `/tmp/watermark.pdf` right away in the first command, which would make sense, then I get a white PDF.

Any clues as to why?

Thanks,
martin

**Comment by urban-warrior** [comment, marked as answer]
Writing to PNG and reading it back it with a separate command modifies the environment, specifically the page value.  Add `-page 2480x3508+0+0` right before the `/tmp/watermark.png` command-line argument.  You should now get expected results.

**Reply by madduck** [reply]
Indeed, that works and produces a PDF. I thought `+repage` would do that, but alas, no.
